{{ cookiecutter.project_name }}/.github/workflows/test.yml

```diff
@@ -5,29 +5,25 @@ on:
   push:
     branches:
       - master
+      - fixes_*
   pull_request:
 
 jobs:
   build_and_test:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
     steps:
-      # Checkout the code.
-      - uses: actions/checkout@v2
-      # Activate python environment
-      - uses: actions/setup-python@v1
+      - uses: actions/checkout@v3
+    
+      - name: Set up Python
+        uses: actions/setup-python@v4
         with:
-          python-version: '3.7'
-      - name: build
+          python-version: '3.10'
+
+      - name: Install python dependencies
         run: |
-          python3 -m venv .
-          bin/pip install -r requirements.txt
-      - name: test
-        run: bin/pytest
-      # Reinout: enable this after setting COVERALLS_REPO_TOKEN as a secret.
-      # Note: only if the repo is public. Oh, and use curly braces.
-      # - name: coverage report
-      #   run: bin/coveralls
-      #   env:
-      #     COVERALLS_REPO_TOKEN: $[[ secrets.COVERALLS_REPO_TOKEN ]]
+          pip install --disable-pip-version-check --upgrade pip setuptools 
+          pip install -e .[test]
```

Comment:
Ik doe normaal `requirements.txt` met daarin die `-e .[test]`, dan kan je daar tools inzetten die je verder niet als dependencies wilt.

Reply:
Oh, en sowieso in een virtualenv (want dat doe je lokaal ook).
Of doe je dit specifiek om een reguliere install te testen? Op zich ook een idee. Ik ben erg gewend alles in sync te houden, maar da's misschien niet nodig/gewenst?

Reply:
Ik denk dat je onderscheid moet maken tussen *applications* en *libraries*.

Ik zie deze cookiecutter als een cookiecutter voor libraries. In dat geval maakt het niet uit of een environment nou een virtualenv is of een systeem environment, of voor mijn part conda. Het moet gewoon werken. Het is conventioneel om dan een matrix van dependencies te doen. Ik heb die ook even uit gridbuilder geport.

En wat betreft requirements: dat is iets voor applications. Een library specificeert zijn requirements in de setup.py. Voor test requirments heb je de `[test]` extra requirements.

Reply:
Da's waar.

Mijn neiging om het commando voor lokaal ontwikkelen hetzelfde te houden als in de automatische tests is omdat adviseurs nogal de neiging hebben om zelf wat te klussen qua vage setup bij het ontwikkelen. En dus wil ik de reguliere ontwikkelomgeving werkend houden.

Maar.... deze setup bevalt me ook wel, dus hou maar zo. Er zit een duidelijke reden achter.